File: ruby/largest-series-product/largest_series_product.rb
class Series
  def initialize(str)
    @serie = str.chars.map(&:to_i)
  end

  def largest_product(n)
    @serie.each_cons(n).map do |cons|
      cons.reduce(1, :*)
    end.max
  end
end

# serie = Series.new('29')
# p serie.serie

File: javascript/largest-series-product/largest-series-product.js
export const largestProduct = (string, length) => {
  if (length < 0 || string.match(/\D/)) throw new Error('Invalid input.');
  if (length === 0) return 1;

  const products = [];
  for (let i = 0; i <= string.length - length; i++) {
    const cons = string.substr(i, length).split('').map(char => parseInt(char, 10));
    products.push(cons.reduce((a, b) => a * b));
  }

  return Math.max(...products);
};
